proxy/pkg/zdmproxy/clusterconn.go

```diff
@@ -361,7 +390,16 @@ func (cc *ClusterConnector) handleAsyncResponse(response *frame.RawFrame) *frame
 }
 
 func (cc *ClusterConnector) sendRequestToCluster(frame *frame.RawFrame) {
-	cc.writeCoalescer.Enqueue(frame)
+	var err error
+	if cc.frameProcessor != nil {
+		frame, err = cc.frameProcessor.AssignUniqueId(frame)
+	}
+	if err != nil {
+		log.Errorf("couldn't assign unique id to frame %v", frame)
+		handleConnectionError(err, cc.clusterConnContext, cc.cancelFunc, string(cc.connectorType), "writing", cc.connection.RemoteAddr().String())
```

Comment:
hmm I'm not sure if calling `handleConnectionError` is the right choice here, this function is usually called by the read/write goroutines when an IO error happens to trigger a shutdown operation on all connections related to this connector (client and other cluster).

In this scenario I think it's better to just log the error and return

Reply:
Done